utils_linux.go

```diff
@@ -171,6 +171,9 @@ func createPidFile(path string, process *libcontainer.Process) error {
 	if err != nil {
 		return err
 	}
+	if _, err := os.Stat(path); err == nil {
+		return fmt.Errorf("file %s already exists", path)
```

Comment:
you can probably detect if the file is a socket using its filemode. However i feel this is a bit overkill considering that this is more or less an user misconfiguration that can be very easily detected much earlier in the code by just comparing the flags, or perhaps even adding some documentation will suffice.

We only have two file flags now, i dont think its very confusing.